### DISCUSSION TITLE
Duration of PgSTAC mosaic search ID

### DISCUSSION THREAD

**Discussion by ebkurtz** [microsoft/PlanetaryComputer#348] [Q&A] [answered]
When I register a search with the PgSTAC mosaic `/api/data/v1/mosaic/register` does the search ID ever expire? I would like to use the XYZ tile endpoint and want to know if I will have to periodically regenerate it with a new search ID.

**Comment by mmcfarland** [comment, marked as answer]
Hi @ebkurtz -- while the search IDs are not _guaranteed_ to persist, they should be considered stable and you won't need to periodically regenerate them. They are deterministically created based on the `register` content, so the exact same query should produce the same hash in the event that we would need to flush them from the system (i.e., client code wouldn't need to update to a new search ID).

**Reply by ebkurtz** [reply]
Thanks for the speedy answer, this is helpful.
